internal/stores/contracts.go

```diff
@@ -179,3 +194,382 @@ func NewJSONContractStore(dir string) (*JSONContractStore, error) {
+	}
+
+	dbSiaPublicKey struct {
+		ID                uint64          `gorm:"primaryKey"`
+		Algorithm         types.Specifier `gorm:"type:bytes;serializer:gob"`
+		Key               []byte
+		UnlockConditionID types.FileContractID `gorm:"index;type:bytes;serializer:gob;NOT NULL"`
+	}
+
+	dbValidSiacoinOutput struct {
+		ID         uint64               `gorm:"primaryKey"`
+		ParentID   types.FileContractID `gorm:"index;type:bytes;serializer:gob;NOT NULL"`
+		UnlockHash types.UnlockHash     `gorm:"index;type:bytes;serializer:gob"`
+		Value      *big.Int             `gorm:"type:bytes;serializer:gob"`
+	}
+
+	dbMissedSiacoinOutput struct {
+		ID         uint64               `gorm:"primaryKey"`
+		ParentID   types.FileContractID `gorm:"index;type:bytes;serializer:gob;NOT NULL"`
+		UnlockHash types.UnlockHash     `gorm:"index;type:bytes;serializer:gob"`
+		Value      *big.Int             `gorm:"type:bytes;serializer:gob"`
+	}
+
+	dbTransactionSignature struct {
+		ID             uint64               `gorm:"primaryKey"`
+		ParentID       types.FileContractID `gorm:"index;type:bytes;serializer:gob;NOT NULL"`
+		PublicKeyIndex uint64
+		Timelock       types.BlockHeight
+		CoveredFields  types.CoveredFields `gorm:"type:bytes;serializer:gob"`
+		Signature      []byte
+	}
+
+	dbHostSet struct {
+		Name  string           `gorm:"primaryKey"`
+		Hosts []dbHostSetEntry `gorm:"constraing:OnDelete:CASCADE;foreignKey:HostSetName;references:Name"`
+	}
+
+	dbHostSetEntry struct {
+		ID          uint64              `gorm:"primaryKey"`
+		HostSetName string              `gorm:"index;NOT NULL"`
+		PublicKey   consensus.PublicKey `gorm:"NOT NULL;type:bytes;serializer:gob"`
+	}
+)
+
+// TableName implements the gorm.Tabler interface.
+func (dbContractRHPv2) TableName() string { return "contracts_v2" }
+
+// TableName implements the gorm.Tabler interface.
+func (dbFileContractRevision) TableName() string { return "file_contract_revisions" }
+
+// TableName implements the gorm.Tabler interface.
+func (dbTransactionSignature) TableName() string { return "transaction_signatures" }
+
+// TableName implements the gorm.Tabler interface.
+func (dbUnlockConditions) TableName() string { return "unlock_conditions" }
+
+// TableName implements the gorm.Tabler interface.
+func (dbSiaPublicKey) TableName() string { return "public_keys" }
+
+// TableName implements the gorm.Tabler interface.
+func (dbValidSiacoinOutput) TableName() string { return "siacoin_valid_outputs" }
+
+// TableName implements the gorm.Tabler interface.
+func (dbMissedSiacoinOutput) TableName() string { return "siacoin_missed_outputs" }
+
+// TableName implements the gorm.Tabler interface.
+func (dbHostSet) TableName() string { return "host_sets" }
+
+// TableName implements the gorm.Tabler interface.
+func (dbHostSetEntry) TableName() string { return "host_set_entries" }
+
+// Contract converts a dbContractRHPv2 to a rhpv2.Contract type.
+func (c dbContractRHPv2) Contract() (rhpv2.Contract, error) {
```

Comment:
not a big fan of this name. Perhaps we can standardize on calling these methods `convert` or `toOriginal` or something. (Also, does it need to be exported?)

Reply:
usually when working with ORMs I think people go for `DTO` and add stuff like `toDTO`, `fromDTO`, `toEntity` and so on. I never really liked that but when done consistently it's quite enjoyable